Improve LifecycleExecutionState parsing.

martijnvg opened the pull request:
This change improves the parsing of LifecycleExecutionState from IndexMetadata custom data
by avoiding containsKey(...) call and in case there is no custom data then return a blank
LifecycleExecutionState instance.

Relates to #77466

martijnvg committed:
Improve LifecycleExecutionState parsing.

This change improves the parsing of LifecycleExecutionState from IndexMetadata custom data
by avoiding containsKey(...) call and in case there is no custom data then return a blank
LifecycleExecutionState instance.

Relates to #77466

elasticmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

original-brownbear commented on x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecycleExecutionState.java:
```diff
@@ -93,8 +93,11 @@ private LifecycleExecutionState(String phase, String action, String step, String
+        if (customData != null && customData.isEmpty() == false) {
+            return fromCustomMetadata(customData);
+        } else {
+            return LifecycleExecutionState.builder().build();
```
NIT: add a constant for this one?

original-brownbear reviewed (approved):
LGTM

martijnvg committed:
added constant for empty state and
removes assertion (existed just for checking if an empty map was provided from other places)

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/elasticsearch/pull/77860)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/77860">](https://github.com/elastic/elasticsearch/pull/77860) |
